### PR TITLE
Add Grafana Integration

### PR DESCRIPTION
**Pull request by snaselj**
# Closes: #208 

## What's Changed

Added Grafana integration from nautobot-plugin-chatops-grafana.

## TODO

- [x] Test migrations.
- [x] Resolve schema-enforcer pip dependency issue.
- [x] Resolve templates issues.
- [ ] Resolve issue with displaying Panel -> Active in Panel list.
- [x] Do not start together with obsolete nautobot-chatops-grafana plugin.
- [x] Align API endpoints with the UI.

### CODE REVIEW

**Comment by smk4664** [comment]
I have some reservations with this conversion, as it is not just ChatOps. The Grafana repo has Nautobot Models, ChatOps and SSoT.

**Comment by snaselj** [comment]
> I have some reservations with this conversion, as it is not just ChatOps. The Grafana repo has Nautobot Models, ChatOps and SSoT.

Do you think, this can be solved by merging diffsync part to nautobot-plugin-ssot instead of this repo?

File: development/Dockerfile [hunk 1]
```diff
@@ -68,11 +68,13 @@ RUN sort poetry_freeze_base.txt poetry_freeze_all.txt | uniq -u > poetry_freeze_
...
 # Install all local project as editable, constrained on Nautobot version, to get any additional
 #   direct dependencies of the app
-RUN pip install -c constraints.txt -e .
+RUN pip install -c constraints.txt -e .[all]
 
 # Install any dev dependencies frozen from Poetry
 # Can be improved in Poetry 1.2 which allows `poetry install --only dev`
 RUN pip install -c constraints.txt -r poetry_freeze_dev.txt
 
+RUN pip install --no-deps schema-enforcer jsonref jmespath
```
Comment (snaselj) [inline development/Dockerfile:77]: This has to be removed.

File: pyproject.toml [hunk 2]
```diff
@@ -18,20 +18,26 @@ include = [
...
 aiohttp = "^3.7.3"
 asgiref = "^3.4.1"
-Markdown = "!=3.3.5"
+diffsync = "^1.3.0"
+isodate = "^0.6.0"
+nautobot = "^1.4.0"
+nautobot-capacity-metrics = "*"
+pydantic = "^1.8.2"
+python = "^3.7"
+# schema-enforcer = "*"
```
Comment (snaselj) [inline pyproject.toml:36]: Resolve dependency issues when running `poetry lock --no-update`

**Comment by smk4664** [comment]
We will also need to document that this conflicts with the nautobot-plugin-chatops-grafana. Since both apps provide the same Models, we cannot have both installed in the same Nautobot instance.

**Comment by snaselj** [comment]
> We will also need to document that this conflicts with the nautobot-plugin-chatops-grafana. Since both apps provide the same Models, we cannot have both installed in the same Nautobot instance.

Should be resolved by https://github.com/nautobot/nautobot-plugin-chatops/pull/200/commits/543c05cea38c499615dcc7ead6509e7f98b4a11f

**Comment by snaselj** [comment]
Rebased on top of develop.

**Comment by snaselj** [comment]
Marked as a draft as it is blocked by schema-enforcer. Once it will be resolved, will resolve conflicts and mark it ready for review.

**Comment by snaselj** [comment]
Merged with the latest develop to resolve conflicts.

**Comment by snaselj** [comment]
There is still some display issue with the `Active` field:

screenshot

The first row has `active=true` but there is no difference visible. @smk4664 Any ideas how to fix that?

Otherwise conflicts and all issues are resolved, ready for review.

**Comment by smk4664** [comment]
> There is still some display issue with the `Active` field:
> 
> screenshot
> 
> The first row has `active=true` but there is no difference visible. @smk4664 Any ideas how to fix that?
> 
> Otherwise conflicts and all issues are resolved, ready for review.

I believe the issue with the active=true is due to the use of ascii characters instead of MDI Icons. This appears to be an issue in the original repo as well, so I will merge and create a bug ticket for this.

**Pull request merged**